libhb/work.c

```diff
@@ -1229,169 +1229,69 @@ static int sanitize_audio(hb_job_t *job)
 static int sanitize_qsv( hb_job_t * job )
 {
 #if HB_PROJECT_FEATURE_QSV
-#if 0 // TODO: re-implement QSV VPP filtering and QSV zerocopy path
-    int i;
-
     /*
-     * XXX: mfxCoreInterface's CopyFrame doesn't work in old drivers, and our
-     *      workaround is really slow. If we have validated CPU-based filters in
-     *      the list and we can't use CopyFrame, disable QSV decoding until a
-     *      better solution is implemented.
+     * When QSV's VPP is used for filtering, not all CPU filters
+     * are supported, so we need to do a little extra setup here.
      */
-    if (hb_qsv_copyframe_is_slow(job->vcodec))
+
+    if(!job->qsv.ctx)
     {
-        if (job->list_filter != NULL)
+        job->qsv.ctx = av_mallocz(sizeof(hb_qsv_context));
```

Comment:
This appears to also be av_mallocz'd in qsv_enc_init.  And it is freed in encqsvClose. Should it only be allocated in work.c?  If so it should also be freed here.

Reply:
Fixed